src/sagemaker/workflow/steps.py

```diff
@@ -105,6 +108,19 @@ def ref(self) -> Dict[str, str]:
         """Gets a reference dict for steps"""
         return {"Name": self.name}
 
+    @staticmethod
+    def _resolve_depends_on(depends_on_list: Union[List[str], List["Step"]]):
```

Comment:
List["Step"] -> List[Step]

Reply:
in py37 in order to infer type to self class, we needs to do `from __future__ import annotations`, and this is not supported in py36. The pythonSDK supports both 37 and 36, so I think we need to stick with string "Step" for a while.

https://stackoverflow.com/questions/33533148/how-do-i-type-hint-a-method-with-the-type-of-the-enclosing-class